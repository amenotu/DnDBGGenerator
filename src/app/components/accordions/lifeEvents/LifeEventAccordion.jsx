import React from 'react';
import { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Accordion, Icon, Segment } from 'semantic-ui-react';

function mapStateToProps(state, { id }) {
  return {
    lifeEvent: state.getIn(['lifeEvents', 'LifeEvents', id]),
  };
}

class LifeEventAccordion extends Component {
  static PropTypes = {
    id: PropTypes.Number,
    lifeEvent: PropTypes.String,
  }

  constructor(props) {
    super(props);

    const { id } = this.props;

    this.state = {
      activeIndex: {
        0: id === 1 ? true : false,
      },
    };
  }

  componentWillReceiveProps(nextProps) {
    const { id } = this.props;
    this.setState({ activeIndex: {
        0: id === 1 ? true : false,
      },
    });
  }

  handleClick = (e, titleProps) => {
    const { activeIndex } = this.state;
    const index = titleProps;
    const newIndex = activeIndex[index] ? false : true;
    activeIndex[index] = newIndex;

    this.setState({ activeIndex });
  }

  render() {
    const { lifeEvent } = this.props;

    const { activeIndex } = this.state;

    return (
      <div>
        <Accordion inverted>
          <Accordion.Title
            active={ activeIndex[0] }
            onClick={ this.handleClick }>
            <Icon name='dropdown' />
            Life Event
          </Accordion.Title>
          <Accordion.Content active={ activeIndex[0] }>
            <p>
              { lifeEvent }
            </p>
          </Accordion.Content>
        </Accordion>
      </div>
    )
  };
}

export default connect(mapStateToProps, null)(LifeEventAccordion);
