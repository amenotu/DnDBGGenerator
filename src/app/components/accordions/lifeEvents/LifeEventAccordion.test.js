import React from 'react';
import LifeEventAccordion from './LifeEventAccordion.jsx';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import Immutable from 'immutable';

const mockInitialState = Immutable.fromJS({
  lifeEvents: {
    LifeEvents: {
      1: 'Some string',
    },
  },
});

const configStore = configureStore();
const mockStore = configStore(mockInitialState);

const mockProps = { id: '1' };

describe('LifeEventAccordion component', () => {
  it('renders correctly', () => {
    const renderedValue = renderer.create(<LifeEventAccordion { ...mockProps } store={ mockStore } />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
