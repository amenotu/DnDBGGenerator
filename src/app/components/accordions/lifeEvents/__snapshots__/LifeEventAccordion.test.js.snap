// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LifeEventAccordion component renders correctly 1`] = `
<div>
  <div
    className="ui inverted accordion"
  >
    <div
      className="title"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="dropdown icon"
      />
      Life Event
    </div>
    <div
      className="content"
    >
      <p>
        Some string
      </p>
    </div>
  </div>
</div>
`;
